### PR TITLE
Fix ethereum adapter & update Dockerfile Rust version to 1.45.2

### PR DESCRIPTION
**Pull request by elpiel**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-FROM rust:1.40 as builder
+FROM rust:1.45.2 as builder
```
Comment (shirshak55) [inline Dockerfile:1]: why no use latest version released yesterday? Or is it substrate issue afaik they has some issue which required 2020-10-05 version. But I think I would wait for latest version of compiler ?

Comment (elpiel) [reply Dockerfile:1]: I was going to update it, however clippy is panicing, see my comment here: https://github.com/AdExNetwork/adex-validator-stack-rust/issues/330#issuecomment-730882481

We don't use substrate in this repo, the issue arouse from compiler regression, this is why we reverted to 1.45.2 in the first place, I just forgot to update the `Dockerfile`

**Comment by elpiel** [comment]
Upgrading to 1.48.0 + some improvements to make `clippy` not `panic!`ing are implemented in #349

File: Dockerfile (hunk 1 above)
Comment (elpiel) [reply Dockerfile:2]: See #349 for the `1.48.0` upgrade

**Pull request closed without merging**